### PR TITLE
LMS-686 - Restrict users from visiting unauthorized pages on FE and update navbar based on the user role

### PR DESCRIPTION
**Pull request by Vali-R**
## Issue Link
https://framgiaph.backlog.com/view/LMS-686

## Definition of Done
- [x] Restrict users from visiting unauthorized pages
- [x] Update navbar based on user role 

## Screenshots

https://github.com/framgia/sph-lms/assets/110364690/167c58b5-7f73-41b3-8bc3-40fb031ef5f2

### CODE REVIEW

**Review by Melanthonyando — APPROVED** [review]
### Approved 👍 

Proof :
[trainee.webm](https://github.com/framgia/sph-lms/assets/110366151/d4a5aa85-d8a1-4a1b-888a-ffe641aa5d76)
[Trainer.webm](https://github.com/framgia/sph-lms/assets/110366151/b3fd4579-206d-4877-92a8-990df2298424)

Iteration: 0
Comment : 0
Testing Time : 1hr

File: client/src/middleware.ts
Comment (ragg09-sunasterisk) [inline client/src/middleware.ts:1]: please move this file inside the middlewares folder and rename it something like `AccessMiddleware` or any name will do as long as its easy to understand

Comment (Vali-R) [reply client/src/middleware.ts:1]: @ragg09-sunasterisk I've tried it work it out but its not quite possible, its a standard for next.js, there should be `middleware.ts` file in the same level as pages to make use of Next.js's middleware

Source:
https://nextjs.org/docs/pages/building-your-application/routing/middleware

**Review by ragg09-sunasterisk — APPROVED** [review]

**Comment by ragg09-sunasterisk** [comment]
Approved 

Iteration: 0
Comment : 1
Testing Time : 15mins

**Pull request merged**